p2p/src/secret_connection.rs

```diff
@@ -184,24 +183,26 @@ impl Handshake<AwaitingAuthSig> {
             .and_then(|key| key.sum)
             .ok_or_else(Error::missing_key)?;
 
+        #[allow(clippy::match_wildcard_for_single_variants)]
         let remote_pubkey = match pk_sum {
             proto::crypto::public_key::Sum::Ed25519(ref bytes) => {
-                ed25519::PublicKey::from_bytes(bytes).map_err(Error::signature)
+                ed25519_consensus::VerificationKey::try_from(&bytes[..])
+                    .map_err(|_| Error::signature())
             },
-            proto::crypto::public_key::Sum::Secp256k1(_) => Err(Error::unsupported_key()),
```

Comment:
I would rather keep the exhaustive match here, to force changes when new key types get added into protobuf.

Also, we're still working directly with proto, not through domain types. Not to be fixed in this PR, though.